scrapy/contracts/__init__.py

```diff
@@ -51,8 +51,11 @@ def from_method(self, method, results):
             # calculate request args
             args, kwargs = get_spec(Request.__init__)
             kwargs['callback'] = method
+            ignored_errors = set()
             for contract in contracts:
                 kwargs = contract.adjust_request_args(kwargs)
+                if contract.name is 'ignore':
```

Comment:
it should be `contract.name == 'ignore'` because in Python it is not guaranteed that equal strings point to the same object